libvast/test/experimental_table_slice.cpp

```diff
@@ -373,6 +373,75 @@ TEST(record batch roundtrip - adding column) {
   CHECK_VARIANT_EQUAL(slice2.at(3, 1, string_type{}), "3"sv);
 }
 
+auto field_roundtrip(const type& t) {
+  const auto& arrow_field = make_experimental_field({"x", t});
+  const auto& restored_t = make_vast_type(arrow_field->type());
+  if (t != restored_t) { // CHECK_EQUAL doesn't cut it
+    fmt::print(stderr, "`{}` != `{}`\n", t, restored_t);
+    fmt::print(stderr, "arrow schema: {}\n", arrow_field->ToString(true));
+  }
+  CHECK_EQUAL(t, restored_t);
+}
+
+TEST(arrow primitive type to field roundtrip) {
+  field_roundtrip(type{none_type{}});
+  field_roundtrip(type{bool_type{}});
+  field_roundtrip(type{integer_type{}});
+  field_roundtrip(type{count_type{}});
+  field_roundtrip(type{real_type{}});
+  field_roundtrip(type{duration_type{}});
+  field_roundtrip(type{time_type{}});
+  field_roundtrip(type{string_type{}});
+  // does not work yet: cannot be distinguished from string
+  // field_roundtrip(type{pattern_type{}});
+  field_roundtrip(type{address_type{}});
+  field_roundtrip(type{subnet_type{}});
+  // currently a value of type count, indistinguishable from a normal count
+  // field_roundtrip(type{enumeration_type{{"first"}, {"third", 2}, {"fourth"}}});
+  field_roundtrip(type{list_type{integer_type{}}});
+  // impossible to distinguish from list_type<stuct<key, value>>:
+  // field_roundtrip(type{map_type{integer_type{}, address_type{}}});
+  field_roundtrip(
+    type{record_type{{"key", integer_type{}}, {"value", address_type{}}}});
+  field_roundtrip(
+    type{record_type{{"a", string_type{}}, {"b", address_type{}}}});
+  field_roundtrip(type{record_type{
+    {"a", string_type{}},
+    {"b", record_type{{"hits", count_type{}}, {"net", subnet_type{}}}}}});
+}
+
+auto schema_roundtrip(const type& t) {
+  const auto& arrow_schema = make_experimental_schema(t);
+  const auto& restored_t = make_vast_type(*arrow_schema);
+  if (t != restored_t) { // CHECK_EQUAL doesn't cut it
```

Comment:
Note: For a better debugging experience, you can define this somewhere:

```
void inspect(caf::detail::stringification_inspector& f, const arrow::Schema& x) {
  auto str = x->ToString(true);
  f(str);
}
```

And similar for the other types. This is the code path that `CHECK_EQUAL` takes for printing things in case something goes wrong.